rasterio/_features.pyx

```diff
@@ -347,10 +353,17 @@ def _bounds(geometry):
             ymins.append(ymin)
             xmaxs.append(xmax)
             ymaxs.append(ymax)
-        return min(xmins), min(ymins), max(xmaxs), max(ymaxs)
+        if north_up:
+            return min(xmins), min(ymins), max(xmaxs), max(ymaxs)
+        else:
+            return min(xmins), max(ymaxs), max(xmaxs), min(ymins)
+
     else:
         xyz = tuple(zip(*list(_explode(geometry['coordinates']))))
-        return min(xyz[0]), min(xyz[1]), max(xyz[0]), max(xyz[1])
+        if north_up:
+            return min(xyz[0]), min(xyz[1]), max(xyz[0]), max(xyz[1])
+        else:
+            return min(xyz[0]), max(xyz[1]), max(xyz[0]), min(xyz[1])
```

Comment:
Absolutely. Mental note to use the `left, bottom, right, top` terminology everywhere from now on.